modules/social_features/social_event/src/Form/EnrollActionForm.php

```diff
@@ -187,10 +193,21 @@ public function buildForm(array $form, FormStateInterface $form_state) {
             break;
           }
 
-          if ($group->hasPermission('enroll to events in groups', $current_user) == FALSE) {
-            return [];
+          if ($enroll_to_events_in_groups) {
+            // Skip permission validation if 'enroll to events in groups' is
+            // already granted.
+            break;
+          }
+          elseif ($group->hasPermission('enroll to events in groups', $current_user) === TRUE) {
+            $enroll_to_events_in_groups = TRUE;
```

Comment:
You can add a `break;` here. Then you can also remove the if statement above, since the foreach will be stopped, once you have set the `$enroll_to_events_in_groups = TRUE;`